### PR TITLE
Fixed `plot_contour`

### PR DESCRIPTION
**Pull request by carefree0910**
## Motivation
When mixture types were used as `categorical`, they could not compare to each other, which would raise errors.

## Description of the changes
I fixed it by converting them to `str` but this solution is nowhere near elegant 😕

### CODE REVIEW

**Comment by crcrpar** [comment]
Could you provide us with minimal reproducible code or a test case?

**Comment by carefree0910** [comment]
> Could you provide us with minimal reproducible code or a test case?

Sure!

```python
import optuna


def objective(trial: optuna.trial.Trial) -> float:
    foo = trial.suggest_categorical("foo", [None, 0])
    if foo is None:
        return 0.0
    return 1.0


study = optuna.create_study(study_name="foo")
study.optimize(objective, n_trials=10)
optuna.visualization.plot_contour(study)
```

**Comment by crcrpar** [comment]
TL; DR: Ah, that pattern may or may not be a corner case though should be supported in the short term.

Could you fix the mypy error?

---

Some personal take on this pull request.

The documentation says the type of `choices` is `Sequence[Union[None, bool, int, float, str]]` as you can see in https://optuna.readthedocs.io/en/stable/reference/generated/optuna.trial.Trial.html?highlight=support_categorical#optuna.trial.Trial.suggest_categorical.

In my understanding, this means that all the elements of `choices` should have the same type.
BUT, as the problem you encounter shows, we have not confirmed this assumption.

So, your patch completely makes sense, though alternatively, we can narrow the supported type of `choices` in runtime in order to avoid the problem you saw. By this, I mean, add the change to `suggest_categorical` to check that `choices` have the same type.

Anyway, my take is as follows. I reckon that idealy we should add the type check of `choices` in runtime to `suggest_categorical`. But, since this change might NOT be backward compatible, I'd like to move this pull request forward.

**Comment by carefree0910** [comment]
Thanks for the detailed reply! I'll try to fix the mypy error, but I'm not familiar with it so I may need to struggle a little bit 🤣

---

And here are some of my thoughts:
1) Narrow down the supported type might dramatically reduce the flexibility of `optuna` 😢.
2) In this case it's weird to see `None` to be one of the supported types, since there could only have one `None` after all 🤣.

**Review by crcrpar — APPROVED** [review]

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the PR. Basically, LGTM. A nit, but should we add the test case when the categorical parameter is used? @carefree0910 What do you think?

**Comment by carefree0910** [comment]
I definitely agree! Just wondering if the test case above is enough and where should I add it to (maybe [here](https://github.com/optuna/optuna/blob/master/tests/visualization_tests/test_contour.py)?) 🤣

**Comment by HideakiImamura** [comment]
@carefree0910 Yes, I think `tests/visualization_tests/test_coutour.py` is fine. Could you add a test?

**Comment by carefree0910** [comment]
Sure!

**Review by HideakiImamura — APPROVED** [review]
Thanks for the update! LGTM!
